File: php/PersonTest.php
<?php

require_once('Person.php');

class PersonTest extends PHPUnit_Framework_TestCase {
  public function setUp(){
    $this->jeff = new Person('Jeff');
    $this->susy = new Person('Susy');
  }

  public function testGetName() {
    $this->assertEquals($this->jeff->getName(), 'Jeff');
  }

  public function testGreetHelloJeff() {
    $greeting = $this->jeff->greet('Hello');
    $this->assertEquals("Hi, I'm Jeff", $greeting);
  }

  public function testGreetHelloSusy() {
    $greeting = $this->susy->greet('Hello');
    $this->assertEquals("Hi, I'm Susy", $greeting);
  }

  public function testGreetGoodbye() {
    $greeting = $this->susy->greet('Goodbye');
    $this->assertEquals('Bye', $greeting);
  }

  public function testGreetOther() {
    $greeting = $this->jeff->greet('Yo');
    $this->assertEquals('Yo', $greeting);
  }
}
?>

File: scala/PersonTest.scala
import org.junit.Assert._
import org.junit.Test
import org.junit.Before

class PersonTest {
  var jeff: Person = _
  var susy: Person = _

  @Before def init() {
    jeff = new Person("Jeff")
    susy = new Person("Susy")
  }

  @Test def testName() {
    assertEquals("Jeff", jeff.name)
  }

  @Test def testGreetHelloJeff() {
    var greeting = jeff.greet("Hello")
    assertEquals("Hi, I'm Jeff", greeting)
  }

  @Test def testGreetHelloSusy() {
    var greeting = susy.greet("Hello")
    assertEquals("Hi, I'm Susy", greeting)
  }

  @Test def testGreetGoodbye() {
    assertEquals("Bye", jeff.greet("Goodbye"))
  }

  @Test def testGreetOther() {
    assertEquals("Yo", jeff.greet("Yo"))
  }
}
